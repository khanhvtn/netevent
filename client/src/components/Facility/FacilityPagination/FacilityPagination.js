import React from 'react';
import { Typography, Select, MenuItem } from '@material-ui/core';
import { Pagination } from '@material-ui/lab';
import { useSelector } from 'react-redux';
//import useStyles in the last
import useStyles from './styles';

const FacilityPagination = ({
    page,
    take,
    handleChangeRowsPerPage,
    handleChangePage,
}) => {
    const css = useStyles();
    const { totalPages } = useSelector((state) => ({
        totalPages: state.facility.totalPages,
    }));
    return (
        <div className={css.paginationWrapper}>
            <div className={css.selectRowNumWrapper}>
                <Typography>Rows per page: </Typography>
                <Select
                    disableUnderline
                    labelId="takeFilterLabel"
                    id="takeFiler"
                    className={css.selectRowNum}
                    variant="standard"
                    value={take}
                    name="take"
                    onChange={handleChangeRowsPerPage}
                >
                    <MenuItem value={5}>5</MenuItem>
                    <MenuItem value={10}>10</MenuItem>
                    <MenuItem value={25}>25</MenuItem>
                </Select>
            </div>
            <div>
                <Pagination
                    page={page}
                    shape="rounded"
                    variant="text"
                    count={parseInt(totalPages)}
                    color="primary"
                    className={css.pagination}
                    onChange={handleChangePage}
                />
            </div>
        </div>
    );
};

export default FacilityPagination;
