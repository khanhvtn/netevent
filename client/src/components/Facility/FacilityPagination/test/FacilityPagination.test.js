import React from 'react'
import { Provider } from 'react-redux'
import { renderer } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import beforeEach from 'jest-wrap'
import expect from 'expect'
import toJson from 'enzyme-to-json'

import FacilityPagination from '../FacilityPagination'

const mockStore = configureStore([])
describe('FacilityPagination', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            myState: 'Facility Pagination'
        })

        store.dispatch = jest.fn()

        component = renderer.create(
            <Provider store={store}>
                <FacilityPagination />
            </Provider>
        )
    })
    
    it('should render with given state from Redux store', () => {
        expect(toJson(component)).toMatchSnapshot()
    })
})