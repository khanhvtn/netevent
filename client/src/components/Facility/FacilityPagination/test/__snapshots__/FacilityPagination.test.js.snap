// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacilityPagination should render with given state from Redux store 1`] = `undefined`;
